packages/core/src/composables/useEditorDecorations.ts

```diff
@@ -1,8 +1,21 @@
-import type { MaybeRefOrGetter } from '@reactive-vscode/reactivity'
-import { toValue, watchEffect } from '@reactive-vscode/reactivity'
+import type { MaybeRef, MaybeRefOrGetter, WatchSource } from '@reactive-vscode/reactivity'
+import { toValue, watch, watchEffect } from '@reactive-vscode/reactivity'
 import type { DecorationOptions, DecorationRenderOptions, Range, TextEditor, TextEditorDecorationType } from 'vscode'
 import { window } from 'vscode'
-import type { Nullable } from '../utils/types'
+import type { Awaitable, Nullable } from '../utils/types'
+import { useDisposable } from './useDisposable'
+import { useDocumentText } from './useDocumentText'
+
+export interface UseEditorDecorationsOptions {
+  /**
+   * The triggers to update the decorations.
+   *
+   * @default ['effect', 'documentChanged']
+   */
+  triggersOn?: ('effect' | 'documentChanged' | WatchSource)[]
```

Comment:
If `trigger` has been changed to `update`, should this also be changed? Like `events`?